src/main/java/net/sf/mzmine/modules/peaklistmethods/io/gnpsexport/GNPSExportTask.java

```diff
@@ -212,10 +226,23 @@ private void export(PeakList peakList, FileWriter writer, File curFile) throws I
         writer.write("CHARGE=" + msmsCharge + msmsPolarity + newLine);
 
         writer.write("MSLEVEL=2" + newLine);
-
-        DataPoint peaks[] = massList.getDataPoints();
-        for (DataPoint peak : peaks) {
-          writer.write(peak.getMZ() + " " + peak.getIntensity() + newLine);
+        boolean merged = false;
+        if (merger != null) {
+          List<MergedSpectrum> spectrum = merger.merge(row, massListName);
+          Optional<MergedSpectrum> max = spectrum.stream().max((u, v) -> Double.compare(v.getBestFragmentScanScore(), u.getBestFragmentScanScore()));
```

Comment:
This looks a bit awkward... I suppose you are checking if the merged spectrum contains any data? 
Why not add a method to MergedSpectrum for that? E.g. ```boolean MergedSpectrum.containsGoodPeaks()```?

Reply:
the previous version of GNPS export was to just export ONE MS/MS (usually, the best one, according to getMostRepresantiveMs2). If you set the merging mode to only merge WITHIN a sample, you will get multiple merged spectra back. Then you decide for the one with best quality.

Reply:
Ok, why not add a method for that like ``merger.getBestMergedSpectrum()```

Reply:
done